library/std/src/sys_common/net.rs

```diff
@@ -688,3 +659,117 @@ impl fmt::Debug for UdpSocket {
         res.field(name, &self.inner.as_raw()).finish()
     }
 }
+
+pub struct UnboundUdpSocket {
+    inner: Socket,
+    addr_family: SocketAddrFamily,
+}
+
+impl UnboundUdpSocket {
+    /// Creates a new UDP socket without binding.
+    pub fn new(addr_family: &SocketAddrFamily) -> io::Result<Self> {
+        init();
+
+        let inner = Socket::new(&addr_family, c::SOCK_DGRAM)?;
+        let addr_family = *addr_family;
+        let new_self = Self { inner,  addr_family };
+        Ok(new_self)
+    }
+
+    pub fn socket(&self) -> &Socket {
+        &self.inner
+    }
+
+    pub fn into_socket(self) -> Socket {
+        self.inner
+    }
+
+    /// Set "SO_REUSEADDR" to true or false. This should be called before
```

Comment:
I think describing what this does is more important than just pushing people off to man pages. Explain what it *does* and then describe how it is implemented.

Reply:
I moved the doc comments with more details into the public API.  And added a link to a stackoverflow article in non-doc comments inside this method.